py/torch_tensorrt/_Input.py

```diff
@@ -180,6 +182,9 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             else:
                 self.torch_tensor = self.example_tensor()
 
+        if "name" in kwargs:
+            self.name = kwargs["name"]
```

Comment:
If the name is not specified, or the user does not know the name (such as in a pretrained model from torchvision, etc.), do we have a way to extract the name or otherwise match the tensor to a name?

Reply:
Additionally, since the `self.name` field is used below, and most users likely will not fill in this field, even when using Dynamic shapes, the script could error out if no default is provided here.

Reply:
> If the name is not specified .....

I had a similar thought. I digged into if we can get input names from the nn.Module but couldn't find one. We can bring this up with Meta and will check the latest source. 

> Additionally, since the self.name ..... 
I added an assertion error now if user has a dynamic input but doesn't provide a non-null name. However, we cannot verify if the name is actually valid since I didn't find anything to compare with.

Reply:
Thanks for the clarification. This sounds good. Additionally, since the `self.name` check happens in the tracer and `self.name` is only defined here if `"name" in kwargs`, then could you also add an `else` here, to set `self.name` to None in the event no name is specified, so that the attribute is still defined? Then the tracer check can just be `if input.name is None`.

Reply:
The name is pre-defined to be an empty string here https://github.com/pytorch/TensorRT/blob/d58bc5df85c49982e6d6c559647ba585541bd38d/py/torch_tensorrt/_Input.py#L50

Reply:
What is the behavior if the user does not provide a name but it is necessary for dynamo?